app/src/org/commcare/connect/network/connectId/parser/ConnectTokenResponseParser.kt

```diff
@@ -21,11 +22,11 @@ class ConnectTokenResponseParser<T>() : BaseApiResponseParser<T> {
                 )
 
                 val json = JSONObject(responseAsString)
-                var key = ConnectConstants.CONNECT_KEY_TOKEN
-                val token = json.getString(key)
+                val token = json.getString(CONNECT_KEY_TOKEN)
+                check(!token.isEmpty() && token != "null") { "$CONNECT_KEY_TOKEN cannot be null or empty" }
+                val seconds = json.optInt(CONNECT_KEY_EXPIRES, 0)
+                check(seconds >= 0) { "$CONNECT_KEY_EXPIRES cannot be negative" }
                 val expiration = Date()
-                key = ConnectConstants.CONNECT_KEY_EXPIRES
-                val seconds = if (json.has(key)) json.getInt(key) else 0
                 expiration.time = expiration.time + (seconds.toLong() * 1000)
```

Comment:
Optional nit: Can we simplify this since we are already here?

```suggestion
                expiration.time += (seconds.toLong() * 1000)
```